private/buf/bufgen/bufgen.go

```diff
@@ -219,7 +220,13 @@ func (p *PluginConfig) GetRemoteHostname() string {
 	if reference, err := bufpluginref.PluginReferenceForString(p.Plugin, 0); err == nil {
 		return reference.Remote()
 	}
-	return p.Remote
+	if p.Remote == "" {
+		return ""
+	}
+	if remote, _, _, _, err := bufremoteplugin.ParsePluginVersionPath(p.Remote); err == nil {
+		return remote
+	}
```

Comment:
Could we log the `err` here, or shown as a warning to the CLI in case the remote is "unparsable"?

Reply:
That should already happen earlier at this location: https://github.com/bufbuild/buf/blob/5d299811bbe0f329b30f3984b92211bf8442d053/private/buf/bufgen/config.go#L206-L212